bin/test-lxd-storage-vm

```diff
@@ -234,6 +234,38 @@ do
         echo "==> Deleting VM"
         lxc delete -f v1
 
+        # Create directory with a file for directory disk tests.
+        mkdir "/tmp/lxd-test-${poolName}"
+
+        # Create empty block file for block disk tests.
+        truncate -s 5m "/tmp/lxd-test-${poolName}/lxd-test-block"
+
+        echo "==> Checking disk device hotplug support"
+        lxc storage volume create "${poolName}" vol1 --type=block size=10MB
+        lxc init images:ubuntu/20.04/cloud v1 --vm -s "${poolName}"
+        lxc start v1
+        waitVMAgent v1
+
+        # Hotplug disks
+        lxc storage volume attach "${poolName}" vol1 v1
+        lxc exec v1 -- stat -c "%F" /dev/sdb | grep "block special file"
+        lxc storage volume detach "${poolName}" vol1 v1
+        ! lxc exec v1 -- stat -c "%F" /dev/sdb || false
+
+        lxc config device add v1 block1 disk source="/tmp/lxd-test-${poolName}/lxd-test-block" readonly=true
+        [ "$(lxc exec v1 -- cat /sys/block/sdb/ro)" -eq 1 ]
+        lxc config device set v1 block1 readonly=false
+        [ "$(lxc exec v1 -- cat /sys/block/sdb/ro)" -eq 0 ]
+
```

Comment:
What happens if we try and attach a filesystem volume to a running VM?

Reply:
```
$ lxc storage volume create default vol2                       
Storage volume vol2 created
$ lxc storage volume attach default vol2 v1 path=/foo
Error: Failed to start device "vol2": Failed to call monitor hook for block device: open 35: no such file or directory
```

Reply:
So we don't support it? If so can we add some validation to make that clear rather than an obscure error that suggests a bug.

Reply:
No, it's not supported. I'll add a clearer message.

Reply:
Cheers